backend/tests/v5/unit/services/filesManager.test.js

```diff
@@ -0,0 +1,195 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { src } = require('../../helper/path');
+const { generateRandomString } = require('../../helper/services');
+
+jest.mock('../../../../src/v5/handler/db');
+const db = require(`${src}/handler/db`);
+jest.mock('../../../../src/v5/models/fileRefs');
+const FileRefs = require(`${src}/models/fileRefs`);
+
+jest.mock('../../../../src/v5/handler/fs');
+const FSHandler = require(`${src}/handler/fs`);
+
+jest.mock('../../../../src/v5/handler/gridfs');
+const GridFSHandler = require(`${src}/handler/gridfs`);
+
+const FilesManager = require(`${src}/services/filesManager`);
+
+const { templates } = require(`${src}/utils/responseCodes`);
+
+const testRemoveAllFilesFromModel = () => {
+	describe('Remove all files from model', () => {
+		test('Should not do any calls if no ref collections are found', async () => {
+			const model = generateRandomString();
+			db.listCollections.mockResolvedValueOnce([
+				`${model}.${generateRandomString()}`,
+				`${model}.${generateRandomString()}`,
+				generateRandomString(),
+			].map((name) => ({ name })));
+			const teamspace = generateRandomString();
+
+			await FilesManager.removeAllFilesFromModel(teamspace, model);
+
+			expect(db.listCollections).toHaveBeenCalledTimes(1);
+			expect(db.listCollections).toHaveBeenCalledWith(teamspace);
+
+			expect(FileRefs.getAllRemovableEntriesByType).not.toHaveBeenCalled();
+		});
+
+		test('Should do the relevant calls to remove files', async () => {
+			const model = generateRandomString();
+
+			const refCol1 = `${model}.${generateRandomString()}.ref`;
+			const refCol2 = `${model}.${generateRandomString()}.ref`;
+
+			db.listCollections.mockResolvedValueOnce([
+				`${model}.${generateRandomString()}`,
+				`${model}.${generateRandomString()}`,
+				refCol1,
+				`${model}.${generateRandomString()}.refasdf`,
+				`${model}.${generateRandomString()}.aaref`,
+				generateRandomString(),
+				refCol2,
+			].map((name) => ({ name })));
+
+			const refCol1Data = [
+				{
+					_id: 'fs',
+					links: [generateRandomString(), generateRandomString(), generateRandomString()],
+				},
+				{
+					_id: 'gridfs',
+					links: [generateRandomString(), generateRandomString(), generateRandomString()],
+				},
+			];
+
+			FileRefs.getAllRemovableEntriesByType
+				.mockImplementation((ts, col) => Promise.resolve(col === refCol1 ? refCol1Data : [{ _id: 'fs', links: [] }]));
+
+			const teamspace = generateRandomString();
+
+			await FilesManager.removeAllFilesFromModel(teamspace, model);
+
+			expect(db.listCollections).toHaveBeenCalledTimes(1);
+			expect(db.listCollections).toHaveBeenCalledWith(teamspace);
+
+			expect(FileRefs.getAllRemovableEntriesByType).toHaveBeenCalledTimes(2);
```

Comment:
Do we wanna test the method calls that are happening outside of the method ? I thought we would only wanna test the calls of removeAllFilesInCol, thats what I do in all unit tests

Reply:
We're not testing that function, that function is mocked. we're just testing that that function is called

Reply:
this is akin to checking db.find() etc are called when we unit test files in the `models` folder